pkg/operator/ceph/cluster/osd/volumes.go

```diff
@@ -106,3 +143,31 @@ func getUdevVolume() (v1.Volume, v1.VolumeMount) {
 
 	return volume, volumeMounts
 }
+
+func getEncryptionVolume(osdID int) (v1.Volume, v1.VolumeMount) {
+	var m int32 = 0400
+	volume := v1.Volume{
+		Name: osdEncryptionVolName,
+		VolumeSource: v1.VolumeSource{
+			Secret: &v1.SecretVolumeSource{
+				SecretName: generateOSDEncryptionSecretName(osdID),
+				Items: []v1.KeyToPath{
+					{
+						Key:  OsdEncryptionSecretNameKeyName,
+						Path: encryptionKeyFileName,
+					},
+				},
+				DefaultMode: &m,
+			},
+		},
+	}
+
+	// Mounts /etc/ceph/luks_key
```

Comment:
Can we mount this to `/etc/ceph`? We are already using that mount path for the ceph config overrides, right?

Reply:
We can, this is only needed for the init activation container so it won't be propagated to the main OSD pod.